tree.go

```diff
@@ -7,30 +7,25 @@ import (
 	"github.com/NebulousLabs/merkletree"
 )
 
-var DefaultTreeCreator = &defaultTreeCreator{}
-
 type Tree interface {
 	Push(data []byte)
 	// TODO(ismail): is this general enough?
 	Prove(idx int) (merkleRoot []byte, proofSet [][]byte, proofIndex uint64, numLeaves uint64)
 	Root() []byte
 }
 
-type TreeCreator interface {
-	New() Tree
-}
-
-var _ TreeCreator = &defaultTreeCreator{}
 var _ Tree = &DefaultTree{}
 
-type defaultTreeCreator struct{}
+// TreeFn creates a fresh Tree instance to be used as the Merkle inside of rsmt2d.
+type TreeFn = func() Tree
```

Comment:
Wonder if TreeInit or TreeConstructor would be a more descriptive name than TreeFn?

Reply:
Renamed to `TreeConstructorFn` in 589b0b3.